### ISSUE TITLE
Import existing hosted zones to an application

### ISSUE THREAD

**Issue by OffensiveBias-08-145** [aws/copilot-cli#3193]
Currently, from my understanding, Copilot automatically creates a new R53 hosted zone. It would be beneficial to allow the import of an existing R53 hosted zone similar to the ability to import an existing VPC.

As an example use case, we have our envs separated into different accounts using AWS SSO with each of our subdomains having their own hosted zone in their respective accounts. 

Instead of copilot creating a new hosted zone and causing conflicts with existing resources, it would be nice to have the ability to import the current Hosted Zone ID.

**Comment by huanjani** [comment]
Hello, @L14MJ4M3S0N! 
[This](https://github.com/aws/copilot-cli/issues/2117) sounds related to your request.
Connecting the two issues so you can follow progress.
Thanks!

**Comment by PedroAlvarado** [comment]
Is there any workaround to be allow import of existing hosted zone?

**Comment by dannyrandall** [comment]
Hey @PedroAlvarado! Yeah, there is.😊 If you [import an ACM certificate to your environment](https://aws.github.io/copilot-cli/docs/developing/domain/#use-domain-in-your-existing-validated-certificates), you can set [`http.hosted_zone`](https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/#http-hosted-zone) in your service manifest with your hosted zone ID, and copilot will create `A` records for your service in there.

**Comment by val1tine** [comment]
@dannyrandall I do not believe that allows us to have a pre-existing hosted zone that can be imported during an app init instead of creating a new one. 
 If I understand it correctly, the `http.hosted_zone` in an svc manifest is only evaluated during `copilot svc init/deploy/etc...`.  When doing an `app delete` and a subsequent `app init`, the hosted zone will be deleted/recreated regardless of what is in the svc manifest.

Ideally, we would be able to do the following:
 -  Manually create pre-existing hosted zone at `<app>.<domain-name>`
 -  Being able to specify that copilot uses this hosted-zone/domain for the app
 -  During `app init`, copilot populates this pre-existing hosted zone with the appropriate NS records
 -  During `svc deploy`, copilot places the appropriate A records in the appropriate hosted zone
 -  During `app delete`, copilot does not delete the hosted zone, but removes the appropriate records from that hosted zone

**Comment by PedroAlvarado** [comment]
@dannyrandall Thank you for your response. Following-up and per @val1tine comments, it seems that when using `app init` you do end-up with a newly create hosted zone. I take there is not way for copilot to simply use an existing hosted zone (for any record) across the board and not create new ones?

**Comment by dannyrandall** [comment]
Hi @PedroAlvarado and @val1tine - sorry for the confusion! Importing an cert to your environment only works at an environment level (meaning you have to import a certificate to each environment, and specify the hosted zone ID to create records in for every service), **not with app init**. There is currently no workaround to do `app init --domain <domain-name>` and have Copilot _not_ create the `<app>.<domain-name>` hosted zone.

To clarify - If the `<domain-name>` hosted zone _already exists in the same account as the account you're running `app init` in_, you can use aliases for your service such that Copilot will create A records in the given hosted zone. For example:
```
# credentials set to account with "mydomain.com" hosted zone
copilot app init --name myapp --domain "mydomain.com"

copilot env init --name myenv # select a different env account and/or region if you'd like
copilot env deploy --name myenv
```

Then, when you create a service, you can specify your aliases using `http.alias` as any of the following:
```
mydomain.com # creates A record in imported hosted zone
service.mydomain.com # creates A record in imported hosted zone
myapp.mydomain.com # creates A record in Copilot-created app hosted zone
service.myapp.mydomain.com # creates A record in Copilot-created app hosted zone 
myenv.myapp.mydomain.com # creates A record in Copilot-created env hosted zone
service.myenv.myapp.mydomain.com # creates A record in Copilot-created env hosted zone
```

**Comment by Lou1415926** [comment]
Hey @PedroAlvarado !  Just a few more question in addition to @dannyrandall's answer.

> Manually create pre-existing hosted zone at `<app>.<domain-name>`

I am curious to know why you'd like to bring `<app>.<domain-name>`  on your own? Is this hosted zone also used by some other tools? 

Or, is it because `<domain-name>` is owned by another account, and you had to delegate `<app>.<domain-name>` to this account (because you couldn't delegate the root `<domain-name>`)?

**Comment by val1tine** [comment]
@Lou1415926 <domain-name> is managed/owned outside of AWS, and we need to add ns records to point to <app>.<domain-name>.  The fact that a new hosted zone is created every `app init` (with new NS records) causes us to have to update our DNS records in that external service.  It would be ideal for us to set up <app>.<domain-name> once, and have it persist across `app init` operations.

**Comment by Lou1415926** [comment]
@val1tine Ah that makes sense to me - super helpful! Thanks for sharing!
